### PR TITLE
Fix editor markdown not incrementing in a numbered list

### LINKED ISSUE #33184: Markdown editor doesn't increment points in numbered lists on enter
### Description

When making a numbered list the editor always inserts `1.` as next item.

Reproduction steps:
1. Make an issue
2. Make a numbered list with more than one point
3. Next point is always `1.`

This has no impact on later rendering of the list, numbers are correct there.
https://demo.gitea.com/TheFox0x7/runs-on_bug/issues/1

### Screenshots

image

### Gitea Version

2298ff2152

### Can you reproduce the bug on the Gitea demo site?

Yes

### Operating System

Linux

### Browser Version

Firefox 133.0.3

### PR DESCRIPTION
**Pull request by harryvince**
Amended the logic for newPrefix in the MarkdownEditor to resolve incorrect number ordering.

Fixes #33184

Attached screenshot of fixed input similar to issue 
Screenshot 2025-01-09 at 23 59 24

### CODE REVIEW

File: web_src/js/features/comp/EditorMarkdown.test.ts [hunk 1]
```diff
@@ -32,10 +32,10 @@ test('EditorMarkdown', () => {
   testInput({value: '1. \n2. ', pos: 3}, {value: '\n2. ', pos: 0});
 
   testInput('- x', '- x\n- ');
-  testInput('1. foo', '1. foo\n1. ');
-  testInput({value: '1. a\n2. b\n3. c', pos: 4}, {value: '1. a\n1. \n2. b\n3. c', pos: 8});
+  testInput('1. foo', '1. foo\n2. ');
+  testInput({value: '1. a\n2. b\n3. c', pos: 4}, {value: '1. a\n2. \n2. b\n3. c', pos: 8});
```
Comment (wxiaoguang) [inline web_src/js/features/comp/EditorMarkdown.test.ts]: The more background details are here:

* https://github.com/go-gitea/gitea/pull/32966#issuecomment-2560499604
* https://github.com/go-gitea/gitea/pull/32966#issuecomment-2560837047

I didn't do that because there will be many edge cases to make the "increment" right. While keep using `1. ` still make the render work right.

----

So here, I am not sure `1 2 2 3` is better or ` 1 1 1 1` is better.

**Comment by wxiaoguang** [comment]
I will try to make a full fix here, to handle all edge cases including:

```
1. a
2. b <- enter here, then last line should be "4. c"
3. c
```

**Comment by wxiaoguang** [comment]
Now most edge cases should have been fixed, and added more tests

**Review by wxiaoguang — APPROVED** [review]
So many edge cases .... I have done my best 🤣

**Review by lunny — APPROVED** [review]
